channels/pkg/channels/apply.go

```diff
@@ -45,89 +39,48 @@ func (p *Applier) Apply(ctx context.Context, manifest []byte) error {
 		return fmt.Errorf("failed to parse objects: %w", err)
 	}
 
-	objectsByGVK := make(map[schema.GroupVersionKind][]*kubemanifest.Object)
-	for _, object := range objects {
-		key := object.GetNamespace() + "/" + object.GetName()
-		gv, err := schema.ParseGroupVersion(object.APIVersion())
-		if err != nil || gv.Version == "" {
-			return fmt.Errorf("failed to parse apiVersion %q in object %s", object.APIVersion(), key)
-		}
-		kind := object.Kind()
-		if kind == "" {
-			return fmt.Errorf("failed to find kind in object %s", key)
-		}
-
-		gvk := gv.WithKind(kind)
-		objectsByGVK[gvk] = append(objectsByGVK[gvk], object)
-	}
-
-	var applyErrors error
-	for gvk := range objectsByGVK {
-		if err := p.applyObjectsOfKind(ctx, gvk, objectsByGVK[gvk]); err != nil {
-			applyErrors = multierr.Append(applyErrors, fmt.Errorf("failed to apply objects of kind %s: %w", gvk, err))
-		}
+	// TODO: Cache applyset for more efficient applying
+	patchOptions := metav1.PatchOptions{
+		FieldManager: "kops",
 	}
-	return applyErrors
-}
 
-func (p *Applier) applyObjectsOfKind(ctx context.Context, gvk schema.GroupVersionKind, expectedObjects []*kubemanifest.Object) error {
-	klog.V(2).Infof("applying objects of kind: %v", gvk)
+	// We force to overcome errors like: Apply failed with 1 conflict: conflict with "kubectl-client-side-apply" using apps/v1: .spec.template.spec.containers[name="foo"].image
+	// TODO: How to handle this better
```

Comment:
It's complicated :-)  I think the short answer is we don't have a choice and have to force eventually.  But we could do something like try first without forcing, log the conflict if there is one, and then force.  This would mean that if there was a loop we could log/detect it.  We could even do things like back-off on the force apply.  I think for an apply library (which this aspires to be) we can do better, but in practice this is the best we can do today!

Reply:
Added this text to the comment :-)